js/iframe-coordinator/src/HostRouter.ts

```diff
@@ -1,97 +1,73 @@
-import ClientFrame from './elements/x-ifc-frame';
-import {
-  ClientToHost,
-  validate as validateIncoming
-} from './messages/ClientToHost';
-import {
-  HostToClient,
-  validate as validateOutgoing
-} from './messages/HostToClient';
-import { Publication } from './messages/Publication';
-
 interface ClientRegistration {
   url: string;
   assignedRoute: string;
 }
 
-class HostRouter {
-  private _routingMap: { [key: string]: ClientRegistration };
-  private _clientFrame: ClientFrame;
-  private _toHostSubscriptions: SubscribeHandler[];
-  private _interestedTopics: Set<string>;
+interface RoutingMap {
+  [key: string]: ClientRegistration;
+}
 
-  constructor(options: {
-    node: HTMLElement;
-    routingMap: { [key: string]: ClientRegistration };
-  }) {
-    this._routingMap = options.routingMap;
-    this._interestedTopics = new Set();
-    this._toHostSubscriptions = [];
+interface ClientInfo extends ClientRegistration {
+  id: string;
+}
 
-    this._clientFrame = new ClientFrame();
-    this._clientFrame.setAttribute('src', 'about:blank');
-    this._clientFrame.addEventListener('clientMessage', (data: CustomEvent) => {
-      const validate = validateIncoming(data.detail);
-      if (validate) {
-        this._clientMessageFromFrame(validate);
-      }
-    });
-    options.node.appendChild(this._clientFrame);
-  }
+class HostRouter {
+  private _clients: ClientInfo[];
 
-  public subscribeToMessages(topic: string): void {
-    this._interestedTopics.add(topic);
+  constructor(clients: RoutingMap) {
+    this._clients = Object.keys(clients).map(id => {
+      return parseRegistration(id, clients[id]);
+    });
   }
 
-  public unsubscribeToMessages(topic: string): void {
-    this._interestedTopics.delete(topic);
+  public getClientUrl(rawRoute: string): string | null {
+    const route = normalizeRoute(rawRoute);
+    this._clients.forEach(client => {
+      if (matchAndStripPrefix(route, client.assignedRoute)) {
+        return applyRoute(client.url, route);
+      }
+    });
+    return null;
   }
+}
 
-  public onSendToHost(handler: SubscribeHandler): void {
-    this._toHostSubscriptions.push(handler);
+function matchAndStripPrefix(
```

Comment:
these will need a private or public in front of them.

Reply:
Do they need that since they are not in the class?  I opted to move stateless helpers out of the class and have them not be exposed by the module, in order to make them actually inaccessible to other modules. People shouldn't rely on private fields or those with the underscore prefix, but I think it's better to just keep as much implementation out of the generated objects as possible.

Reply:
oh, no.  If its out of the class its fine.  I guess i misread the diff.  I am still a noob at github :)

Reply:
Github diffs are pretty bad.